functionary/builder/utils.py

```diff
@@ -224,11 +222,18 @@ def build_package(build_id: UUID):
     with transaction.atomic():
         # Build has succeeded, save all the things now
         if build.status == Build.COMPLETE:
-            for func in db_functions:
+            for func in functions:
                 func.save()
+
+            for parameter in function_parameters:
+                parameter.save()
+
             package.image_name = image_name
             package.save()
 
+            _delete_removed_function_parameters(function_parameters, package)
+            _deactivate_removed_functions(function_definitions, package)
```

Comment:
I realized this `deactivate_removed_functions` call was in the wrong spot before, so I fixed that here.